subsys/shell/modules/kernel_service.c

```diff
@@ -218,6 +222,67 @@ static int cmd_kernel_stacks(const struct shell *shell,
 }
 #endif
 
+static int cmd_kernel_sleep(const struct shell *sh,
+			    size_t argc, char **argv)
+{
+	ARG_UNUSED(sh);
+	ARG_UNUSED(argc);
+
+	k_msleep(strtoul(argv[1], NULL, 10));
+
+	return 0;
+}
+
+#if defined(CONFIG_LOG_RUNTIME_FILTERING)
+static uint32_t log_source_id_get(const char *name)
+{
+	uint32_t count = log_src_cnt_get(CONFIG_LOG_DOMAIN_ID);
+
+	for (uint32_t i = 0; i < count; i++) {
+		if (strcmp(log_source_name_get(CONFIG_LOG_DOMAIN_ID, i), name)
+		    == 0) {
+			return i;
+		}
+	}
+
+	/* Return an invalid source ID */
+	return UINT32_MAX;
+}
+
+static int cmd_kernel_log_level_set(const struct shell *sh,
+				    size_t argc, char **argv)
+{
+	ARG_UNUSED(argc);
+	ARG_UNUSED(argv);
+	int err = 0;
+
+	uint8_t severity = shell_strtoul(argv[2], 10, &err);
+
+	if (err) {
```

Comment:
Don't the `shell_*()` APIs promises to set `err` to 0 in the case of success? I think that'd be cleaner than forcing the caller to pre-initialise `err` always to 0 before the call.

Reply:
I agree, and found it odd too. @Andrewpini What was the reasoning for not setting `*err = 0` in case of success?

Reply:
I think it was to prevent bloating inside modules where you need to do this check alot. 

This is one example from mesh_shell:
```
static int hb_sub_set(const struct shell *shell, size_t argc, char *argv[])
{
	struct bt_mesh_cfg_hb_sub sub;
	uint8_t status;
	int err = 0;

	sub.src = shell_strtoul(argv[1], 0, &err);
	sub.dst = shell_strtoul(argv[2], 0, &err);
	sub.period = shell_strtoul(argv[3], 0, &err);
	if (err) {
		shell_warn(shell, "Unable to parse input string argument");
		return err;
	}

...
```

Here we can merge error checking for all three incoming arguments. If you set the err value to zero you would need a separate check for each parameter. In the mesh shell module we call the `shell_strtoul()` approxamatley 119 times so it would require a significant amount of checks.

Reply:
It is more or less like errno is used in strtoul and strtol.

Reply:
Thanks :) 

So to answer @jhedberg question: The shell commands do not set `err` in case of success, and thus this initialization is required.

Reply:
Right 👍